### PR TITLE
Dev appplatform microsoft.app platform 2020 11 01 preview

### PR DESCRIPTION
**Pull request by superrdean**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json [hunk 1]
```diff
@@ -4014,6 +4018,20 @@
         }
       }
     },
+    "ResourceRequests": {
+      "description": "Deployment resource request payload",
+      "type": "object",
+      "properties": {
+        "cpu": {
```
Comment (filizt) [inline specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json:4025]: >  "cpu": { [](start = 6, length = 10)

As the property name this is not clear how the unit should be. I'd recommend making the name a bit clear and changing the type to integer. Please see an example of how both cpu and memory can be defined here: [https://github.com/Azure/azure-rest-api-specs/blob/abfe5f82e8ac29a05070957bc828874a625079f1/specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json#L7609](https://github.com/Azure/azure-rest-api-specs/blob/abfe5f82e8ac29a05070957bc828874a625079f1/specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json#L7609)

**Comment by filizt** [comment]
>           "version": "1.0",

The version should be the API version in review

---
Refers to: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/examples/Deployments_CreateOrUpdate.json:8 in ae88e43. [](commit_id = ae88e43b14a3d9b006e52c1920946e82d42471a6, deletion_comment = False)

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/examples/Deployments_CreateOrUpdate.json [hunk 2]
```diff
@@ -9,8 +9,10 @@
           "artifactSelector": "sub-module-1"
         },
         "deploymentSettings": {
-          "cpu": 1,
-          "memoryInGB": 3,
```
Comment (filizt) [inline specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/examples/Deployments_CreateOrUpdate.json:13]: Is the removing of these properties intentional? I still see them in the deploymentSettings response model.

**Comment by filizt** [comment]
The PR has breaking changes, please go through the breaking changes review separately from ARM API review. Instructions for breaking change review should be added by the bot above.

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json (hunk 1 above)
Comment (neuqlz) [reply specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json:4025]: It is by design, we want to make cpu and memory field in our API align with the AKS Resource requests convention: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/

For cpu, the user can pass 1 or 1000m to represent the 1c.
For memory, the user can pass 1024Mi or 1Gi to represent the 1Gb.

For now, our system needs to support the size like 0.5c/0.5Gb App, the previous cpu and memoryInGb properties are integer type, so they can't represent the 0.5c/0.5Gb size, so we introduce a new field resourceRequests into the API request.

```
properties: {
  cpu: string
  memoryInGb: string
  resourceRequests: {
    cpu: string
    memory: string
  }
}
```
The user can use ethier cpu/memory or resourceRequests to represent his App's size.

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/examples/Deployments_CreateOrUpdate.json (hunk 2 above)
Comment (neuqlz) [reply specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/examples/Deployments_CreateOrUpdate.json:13]: Yep, it is intentional. According to current design, the users can use either cpu/memory or resourceRequests to represent his App's size. It is a straight either or, they don't need to set them all.

**Comment by neuqlz** [comment]
As for now, we decide to do the change in a new preview version, so close this PR right now, please refer to this PR: https://github.com/Azure/azure-rest-api-specs/pull/13256

Sorry for the disturbing.

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json (hunk 1 above)
Comment (filizt) [reply specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json:4025]: I see - thanks for the explanation.

**Pull request closed without merging**